iModelCore/ECDb/Tests/NonPublished/SchemaUpgradeTests.cpp

```diff
@@ -13542,15 +13542,17 @@ TEST_F(SchemaUpgradeTestFixture, Formats)
                                     <Format typeName="MyFormat" displayLabel="My Format" roundFactor="0.3" type="Fractional" showSignOption="OnlyNegative" formatTraits="TrailZeroes|KeepSingleZero"
                                             precision="4" decimalSeparator="." thousandSeparator="," uomSeparator=" ">
                                         <Composite>
+                                            <Unit>u:KM</Unit>
                                             <Unit label="m">MyMeter</Unit>
+                                            <Unit label="">u:CM</Unit>
                                             <Unit label="mm">u:MM</Unit>
                                         </Composite>
                                     </Format>
                                 </ECSchema>)xml")));
 
     assertFormat(m_ecdb, "MyFormat", "My Format", "",
                  JsonValue(R"json({"roundFactor":0.3, "type": "Fractional", "showSignOption": "OnlyNegative", "formatTraits": ["trailZeroes", "keepSingleZero"], "precision": 4, "decimalSeparator": ".", "thousandSeparator": ",", "uomSeparator": " "})json"),
-                 JsonValue(R"json({"includeZero":true, "units": [{"name":"MyMeter", "label":"m"}, {"name":"MM", "label":"mm"}]})json"));

```

Comment:
I think we should have a test that verifies we can go from empty string to unset and unset to empty string in a schema update.  Making KM empty string and CM unset in the upgraded schema would test that

Reply:
Done !